packages/runtime/src/sdk.rs

```diff
@@ -89,7 +88,63 @@ pub fn link(store: &mut Store<Data>, instance: &mut Instance<Data>) -> anyhow::R
         },
     )?;
 
+    instance.link_closure(
+        &mut *store,
+        "hydrozoa",
+        "getByteArrayPointer",
+        |mut ctx, (address, size): (i32, i32)| {
+            let teavm = ctx.data().teavm.clone().unwrap();
+            let array_ptr = (teavm.byte_array_data)(ctx.as_context_mut(), address).unwrap();
+            let memory = ctx.memory_mut();
+            let slice = &mut memory[array_ptr as usize..(array_ptr as usize + size as usize)];
+
+            let magic = unsafe {
+                // SAFETY: V5_DeviceType is a repr(transparent) struct holding a u8
+                core::mem::transmute::<*mut u8, *mut V5_DeviceType>(slice.as_mut_ptr())
+            };
+            Ok(magic as i32)
```

Comment:
What's going on here? V5_DeviceType seems pretty arbitrary. This whole code just seems equivalent to `memory.as_mut_ptr().wrapping_offset(array_ptr)`

Reply:
I'm not super familiar with the Rust api for dealing with memory, so I looked at other parts of Hydrozoa that use it. I did just have `slice.as_mut_ptr()`, but I was getting a weird 12 byte offset, so I just copied that part of the code, thinking that `transmute` had something to do with it. Your suggestion is better.